client/packages/programs/src/JsonForms/components/BMI.tsx

```diff
@@ -6,26 +6,87 @@ import {
   uiTypeIs,
 } from '@jsonforms/core';
 import { withJsonFormsControlProps } from '@jsonforms/react';
-import { DetailInputWithLabelRow, NumUtils } from '@openmsupply-client/common';
-import { FORM_LABEL_WIDTH } from '../common';
+import {
+  BasicTextInput,
+  Box,
+  DetailInputWithLabelRow,
+  NumUtils,
+  ObjUtils,
+  Typography,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { FORM_LABEL_WIDTH, JsonData, useZodOptionsValidation } from '../common';
+import { z } from 'zod';
+import { useEncounter, useProgramEvents } from '../../api';
 
 export const bmiTester = rankWith(10, uiTypeIs('BMI'));
 
 const round = (value: number) => Math.round(value * 100) / 100;
 
+const Options = z
+  .object({
+    /**
+     * Event type to check for "height" value. If not provided, it will always
+     * retrieve from "height" property of current form data.
+     */
+    eventType: z.string().optional(),
+  })
+  .strict();
+type Options = z.infer<typeof Options>;
+
+const usePreviousHeight = (
```

Comment:
nit: just call it `useHeight` because it also gets current height?

Reply:
029c38e9d